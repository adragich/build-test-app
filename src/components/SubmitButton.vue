<template>
  <button
    type="button"
    class="build-test__button"
    @click="$emit('submit')"
  >
    <span class="build-test__button-text">
      Let it go...
    </span>
  </button>
</template>

<script>
export default {
  name: 'SubmitButton',
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
  .build-test__button {
  	background: #E91E63;
  	border: none;
  	padding: 0;
    width: 280px;
    max-width: 100%;
    box-shadow: 0 4px 2px;
  	border-radius: 8px;
  }
  .build-test__button-text {
  	font-size: 16px;
  	padding: 16px;
  	color: white;
  	display: inline-block;
    width: 100%;
  }
</style>
