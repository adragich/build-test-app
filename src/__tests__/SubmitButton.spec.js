import { shallowMount } from '@vue/test-utils';
import SubmitButton from '../components/SubmitButton.vue';

describe('TextInput.vue', () => {
  const wrapper = shallowMount(SubmitButton);
  
  test('emitting "submit" event on click', () => {
    wrapper.find('button').trigger('click');
    
    expect(wrapper.emitted('submit')).toBeTruthy();
  });

  test('renders button correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
