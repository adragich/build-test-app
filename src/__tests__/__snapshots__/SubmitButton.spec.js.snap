// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput.vue renders button correctly 1`] = `
<button
  class="build-test__button"
  type="button"
>
  <span
    class="build-test__button-text"
  >
    
    Let it go...
  
  </span>
</button>
`;
